src/amo/components/GuidesAddonCard/index.js

```diff
@@ -0,0 +1,218 @@
+import { withInstallHelpers } from 'core/installAddon';
+import translate from 'core/i18n/translate';
+import { getAddonByGUID } from 'core/reducers/addons';
+import { getVersionById } from 'core/reducers/versions';
+import { getErrorMessage } from 'core/utils/addons';
+import { getClientCompatibility } from 'core/utils/compatibility';
+import Card from 'ui/components/Card';
+import Icon from 'ui/components/Icon';
+import Button from 'ui/components/Button';
+import Notice from 'ui/components/Notice';
+import type { AddonType } from 'core/types/addons';
+import type { AddonVersionType } from 'core/reducers/versions';
+import type { AppState } from 'amo/store';
+import type { UserAgentInfoType } from 'core/reducers/api';
+import type { I18nType } from 'core/types/i18n';
+import type { ReactRouterLocationType } from 'core/types/router';
+import type { WithInstallHelpersInjectedProps } from 'core/installAddon';
+
+import './styles.scss';
+
+type Props = {
+  addonGuid: string,
+  addonCustomText: string,
+  staffPick?: boolean,
+};
+
+type InternalProps = {
+  ...Props,
+  ...WithInstallHelpersInjectedProps,
+  _getClientCompatibility: typeof getClientCompatibility,
+  addon: AddonType,
+  clientApp: string,
+  currentVersion: AddonVersionType,
+  defaultInstallSource: string,
+  installError: string,
+  installStatus: string,
+  i18n: I18nType,
+  location: ReactRouterLocationType,
+  userAgentInfo: UserAgentInfoType,
+};
+
+export class GuidesAddonCardBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    _getClientCompatibility: getClientCompatibility,
+    staffPick: true,
+  };
+
+  renderInstallError() {
+    const { i18n, installError: error } = this.props;
+
+    if (!error) {
+      return null;
+    }
+
+    return (
+      <Notice className="Addon-header-install-error" type="error">
+        {getErrorMessage({ i18n, error })}
+      </Notice>
+    );
+  }
+
+  render() {
+    const {
+      _getClientCompatibility,
+      addon,
+      clientApp,
+      currentVersion,
+      i18n,
+      staffPick,
+      userAgentInfo,
+    } = this.props;
+
+    let compatible = false;
+    let compatibility;
+
+    if (addon) {
+      compatibility = _getClientCompatibility({
+        addon,
+        clientApp,
+        currentVersion,
+        userAgentInfo,
+      });
+
+      compatible = compatibility.compatible;
+    }
+
+    const isFireFox =
+      compatibility && compatibility.reason !== INCOMPATIBLE_NOT_FIREFOX;
+    const showInstallButton = addon && isFireFox;
+
+    // TODO: waiting to hear back on how to handle "+Addon" buttons on non
+    // firefox browsers. See:
+    // https://github.com/mozilla/addons-frontend/issues/6432#issuecomment-433083079.
+    const showGetFirefoxButton = addon && !isFireFox;
+
+    return addon ? (
+      <Card>
+        {this.renderInstallError()}
+        <div className="GuidesAddonCard">
+          {isFireFox && !compatible && compatibility ? (
+            <AddonCompatibilityError
```

Comment:
It looks like you can expand `mapStateToProps` within `AddonCompatiblityError` so that you don't have to repeat all this logic (and tests) between here and `pages/Addon`. The final call would look like: `<AddonCompatibilityError addon={addon} />`. Everything else can be mapped from state.

Reply:
hmm..let me check..

Reply:
Again, this could be done in a follow-up issue

Reply:
REF: #6903